### PR TITLE
Cloud Signing

### PR DESCRIPTION
**Pull request by payneBrandon**
Adding remote signing capabilities. Includes MTLS authentication, and new docker-compose.yml file to show usage. Remains backwards compatible with on-prem hosted signing.

### CODE REVIEW

File: sample.env [hunk 1]
```diff
@@ -0,0 +1,8 @@
+SEC_CRYPTO_SERVICE_BASE_URI=
+#The REST endpoint path of the xternal service if `sec.useHsm=false`
```
Comment (dmccoystephenson) [inline sample.env]: Was this supposed to say 'external'?

Comment (payneBrandon) [reply sample.env]: Yep, good catch! Fixed.

File: README.md [hunk 2]
```diff
@@ -1,14 +1,36 @@
 
 # jpo-security-svcs
-This module expopsed a RESTful API for performing cryptographic functions. The following paths identify the functions:
+This module exposes a RESTful API for performing cryptographic functions. The cryptographic functions may be carried out against an on-prem instance, or against a remote instance. Note that for the remote signing use case, MTLS will be used to secure the communications. This requires some additional work on the part of configuration. The following paths identify the functions:
```
Comment (dmccoystephenson) [inline README.md]: This might flow a bit better if it was broken up between the description of the project and the addendum on cryptographic functions. Something like:

> This module exposes a RESTful API for performing cryptographic functions.
> 
> The cryptographic functions may be carried out against an on-prem instance, or against a remote instance. Note that for the remote signing use case, MTLS will be used to secure the communications. This requires some additional work on the part of configuration. The following paths identify the functions:

File: README.md [hunk 3]
```diff
@@ -1,14 +1,36 @@
...
 |--|--|--|--|--|--|
 |POST|/sign|application/json|signs data provided in the body of the request|{"message":"Base64 encoded unsigned data"}|{"result": "Base64 Encoded Signed Data"}
 
+## Mutual TLS Authentication
+For enhanced security, MTLS is used to communicate with a remote signing system. This requires properly configured certificates from the remote system to property perform - work with your remote signing authority to obtain necessary certificates. 
+
+### Certificate Conversion
+Certificates can be in several different formats, including the widely used PEM format. Because this is a Java application, the signing certificates must be in the Java Keystore format to be used. The following commands using [OpenSSL](https://www.openssl.org/) and the Java keytool will convert a PEM file to a Java Keystore file.
+
+First we convert the PEM file to a PKCS12 file:
```
Comment (dmccoystephenson) [inline README.md]: Making this a list of steps rather than using "first" and "then" would make it more readable.

Example:

1. Convert the PEM file to a PKCS12 file:
```
openssl pkcs12 -export -in <cert.pem> -inkey <key.pem> -out <certificate.p12> -name <alias>
```
2. Convert the PKCS12 file to a Java Keystore file:
```
keytool -importkeystore -srckeystore <certificate.p12> -srcstoretype pkcs12 -destkeystore <cert.jks>
```

**Review by dmccoystephenson — APPROVED** [review]
Looks good! I utilized the docker-compose.yml file when testing, successfully sent a request to the application via Postman and received a signed message from the external service.

A short section in the README explaining what we're trying to accomplish in layman's terms would be useful for developers who are unfamiliar with certificate authorities, signing, etc.

File: README.md (hunk 3 above)
Comment (payneBrandon) [reply README.md]: Updated

File: README.md (hunk 2 above)
Comment (payneBrandon) [reply README.md]: Tweaked it to pull the addendum out to a bottom paragraph

Comment (dmccoystephenson) [reply README.md]: I like it!

**Review by drewjj — APPROVED** [review]
Everything looks good
